DataFormats/Detectors/CPV/include/DataFormatsCPV/Cluster.h

```diff
@@ -66,16 +74,47 @@ class Cluster
   char getNExMax() const { return mNExMax; } // Number of maxima found in cluster in unfolding:
                                              // 0: was no unfolging, -1: unfolding failed
 
+  // raw access for CTF encoding
+  uint16_t getPackedPosX() const { return uint16_t((mLocalPosX - kMinX) / kStepX); }
+  void setPackedPosX(uint16_t v) { mLocalPosX = kMinX + kStepX * v; }
+
+  uint16_t getPackedPosZ() const { return uint16_t((mLocalPosZ - kMinZ) / kStepZ); }
+  void setPackedPosZ(uint16_t v) { mLocalPosZ = kMinZ + kStepZ * v; }
+
+  uint8_t getPackedEnergy() const { return uint8_t(std::min(255, int(mEnergy / kStepE))); }
+  void setPackedEnergy(uint16_t v) { mEnergy = v * kStepE; }
+
+  uint8_t getPackedClusterStatus() const
+  {
+    CluStatus s = {0};
+    s.multiplicity = mMulDigit;
+    s.module = mModule;
+    s.unfolded = mNExMax > 1;
+    return s.mBits;
+  }
+  void setPackedClusterStatus(uint8_t v)
+  {
+    CluStatus s = {v};
+    mMulDigit = s.multiplicity;
+    mModule = s.module;
+    mNExMax = s.unfolded ? 1 : 2;
+  }
+
+  void setPacked(uint16_t posX, uint16_t posZ, uint8_t en, uint8_t status)
+  {
+    setPackedPosX(posX);
+    setPackedPosZ(posZ);
+    setPackedEnergy(en);
+    setPackedClusterStatus(status);
+  }
+
  protected:
   char mMulDigit = 0;    ///< Digit nultiplicity
   char mModule = 0;      ///< Module number
   char mNExMax = -1;     ///< number of (Ex-)maxima before unfolding
   int mLabel = -1;       ///< Ref to entry in MCTruthContainer with list of labels
```

Comment:
I see you are setting this label but don't see where it is used. The same for PHOS clusters. Do you need them? Usually we extract MC labels from the container using the cluster index in its container.

Reply:
Thanks, this is rudiment, removed.